File: api/router/game/schemas.py
from pydantic import BaseModel


class InfoIn(BaseModel):
    access_token: str


class InfoOut(BaseModel):
    player: dict
    current_card: dict

    class Config:
        orm_mode = True


class ResultIn(BaseModel):
    bet: int
    prediction: int
    access_token: str


class ResultOut(BaseModel):
    drawn_card: dict
    win: bool
    is_player_bankrupt: bool
    is_deck_empty: bool

    class Config:
        orm_mode = True

File: frontend/src/schemas/schemas.ts
import { z } from 'zod'

export const RegisterResponseSchema = z.object({
    access_token: z.string()
})

export const LoginResponseSchema = z.object({
    access_token: z.string()
})

const Player = z.object({
    name: z.string(),
    credits: z.number(),
})

const Card = z.object({
    sort_index: z.number(),
    rank: z.number(),
    suit: z.number()
})

export const InfoResponseSchema = z.object({
    player: Player,
    current_card: Card
})

export const ResultResponseSchema = z.object({
    drawn_card: Card,
    win: z.boolean(),
    is_player_bankrupt: z.boolean(),
    is_deck_empty: z.boolean()
})